src/overlays/actors/ovl_En_Peehat/z_en_peehat.c

```diff
@@ -92,85 +99,992 @@ static ColliderJntSphInit D_80AD2790 = {
+        { 0xFFCFFFFF, 0x00, 0x00 },
+        TOUCH_ON | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_NONE,
+    },
+    { { { 0.0f, 0.0f, 0.0f }, { 0.0f, 0.0f, 0.0f }, { 0.0f, 0.0f, 0.0f }, { 0.0f, 0.0f, 0.0f } } },
+};
+
+typedef enum {
+    DMG_EFF_ATTACK = 0,
+    DMG_EFF_LIGHT_ICE_ARROW = 6,
+    DMG_EFF_FIRE = 12,
+    DMG_EFF_HOOKSHOT = 13,
+    DMG_EFF_BOOMERANG = 14,
+    DMG_EFF_NUT = 15,
+} DamageEffect;
+
+static DamageTable sDamageTable = {
+    /* Deku nut      */ DMG_ENTRY(0, DMG_EFF_NUT),
+    /* Deku stick    */ DMG_ENTRY(2, DMG_EFF_ATTACK),
+    /* Slingshot     */ DMG_ENTRY(1, DMG_EFF_ATTACK),
+    /* Explosive     */ DMG_ENTRY(2, DMG_EFF_ATTACK),
+    /* Boomerang     */ DMG_ENTRY(0, DMG_EFF_BOOMERANG),
+    /* Normal arrow  */ DMG_ENTRY(2, DMG_EFF_ATTACK),
+    /* Hammer swing  */ DMG_ENTRY(2, DMG_EFF_ATTACK),
+    /* Hookshot      */ DMG_ENTRY(2, DMG_EFF_HOOKSHOT),
+    /* Kokiri sword  */ DMG_ENTRY(1, DMG_EFF_ATTACK),
+    /* Master sword  */ DMG_ENTRY(2, DMG_EFF_ATTACK),
+    /* Giant's Knife */ DMG_ENTRY(4, DMG_EFF_ATTACK),
+    /* Fire arrow    */ DMG_ENTRY(4, DMG_EFF_FIRE),
+    /* Ice arrow     */ DMG_ENTRY(2, DMG_EFF_ATTACK),
+    /* Light arrow   */ DMG_ENTRY(2, DMG_EFF_ATTACK),
+    /* Unk arrow 1   */ DMG_ENTRY(2, DMG_EFF_ATTACK),
+    /* Unk arrow 2   */ DMG_ENTRY(2, DMG_EFF_ATTACK),
+    /* Unk arrow 3   */ DMG_ENTRY(2, DMG_EFF_ATTACK),
+    /* Fire magic    */ DMG_ENTRY(3, DMG_EFF_FIRE),
+    /* Ice magic     */ DMG_ENTRY(0, DMG_EFF_LIGHT_ICE_ARROW),
+    /* Light magic   */ DMG_ENTRY(0, DMG_EFF_LIGHT_ICE_ARROW),
+    /* Shield        */ DMG_ENTRY(0, DMG_EFF_ATTACK),
+    /* Mirror Ray    */ DMG_ENTRY(0, DMG_EFF_ATTACK),
+    /* Kokiri spin   */ DMG_ENTRY(1, DMG_EFF_ATTACK),
+    /* Giant spin    */ DMG_ENTRY(4, DMG_EFF_ATTACK),
+    /* Master spin   */ DMG_ENTRY(2, DMG_EFF_ATTACK),
+    /* Kokiri jump   */ DMG_ENTRY(2, DMG_EFF_ATTACK),
+    /* Giant jump    */ DMG_ENTRY(8, DMG_EFF_ATTACK),
+    /* Master jump   */ DMG_ENTRY(4, DMG_EFF_ATTACK),
+    /* Unknown 1     */ DMG_ENTRY(0, DMG_EFF_ATTACK),
+    /* Unblockable   */ DMG_ENTRY(0, DMG_EFF_ATTACK),
+    /* Hammer jump   */ DMG_ENTRY(4, DMG_EFF_ATTACK),
+    /* Unknown 2     */ DMG_ENTRY(0, DMG_EFF_ATTACK),
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Peehat/func_80ACF4A0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Peehat/EnPeehat_Init.s")
+typedef enum {
+    PEAHAT_STATE_DYING,
+    PEAHAT_STATE_EXPLODE,
+    PEAHAT_STATE_3 = 3,
+    PEAHAT_STATE_4,
+    PEAHAT_STATE_FLY,
+    PEAHAT_STATE_ATTACK_RECOIL = 7,
+    PEAHAT_STATE_8,
+    PEAHAT_STATE_9,
+    PEAHAT_STATE_LANDING, // State landing
+    PEAHAT_STATE_RETURN_HOME = 12,
+    PEAHAT_STATE_STUNNED,
+    PEAHAT_STATE_SEEK_PLAYER,
+    PEAHAT_STATE_15,
+} PeahatState;
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(targetArrowOffset, 700, ICHAIN_STOP),
+};
+
+void EnPeehat_SetActionFunc(EnPeehat* this, EnPeehatActionFunc actionFunc) {
```

Comment:
The current naming convention is to call this type of function `ActorName_SetupAction`.